pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -205,222 +270,296 @@ def handle_sigint(_sig, _frame):
-                True, False, lldb.SBCommandInterpreterRunOptions(), 0, False, False
-            )
-            continue
 
-        # There are interactive commands that `SBDebugger.HandleCommand` will
-        # silently ignore. We have to implement them manually, here.
-        if "quit".startswith(line):
-            break
-        if line == "exit":
-            break
+            last_command = action._command
 
-        # `script` is a little weird. Unlike with the other commands we're
-        # emulating, we actually need LLDB to spawn it for it to make sense
-        # from the perspective of the user. This means we have to make
-        # special arrangements for it.
-        #
-        # There is a way to get LLDB to properly handle interactive commands,
-        # and that is to start an interactive session with
-        # `SBDebugger.RunCommandInterpreter`, but that comes with its own
-        # challenges:
-        #     (1) Starting an interactive session on standard input is the
-        #         best option from the perspective of the user, as they get
-        #         full access to the Python interpreter's readline functions.
-        #         However, we can't start a session running a command, which
-        #         means we open up the possibility of the user breaking
-        #         Pwndbg completely if they type in any process or target
-        #         management commands.
-        #     (2) Setting an input file up for the debugger to use, having
-        #         that input file start the Python interpreter, and piping
-        #         `sys.stdin` to it while the interpreter is running. This
-        #         option is better in that it avoids the possibility of the
-        #         user breaking Pwndbg by mistake, but it breaks both
-        #         readline and input in general for the user.
-        #
-        # While neither option is ideal, both can be partially mitigated.
-        # Option (1) by adding an extra command that drops down to LLDB and
-        # prints a warning to make the user aware of the risk of breaking
-        # Pwndbg, and option (2) by making a TextIOBase class that uses input()
-        # at the REPL level before piping that to the Python interpreter running
-        # under LLDB.
-        #
-        # Currently, we go with the mitigated version of option (1), but option
-        # (2) might still be on the table for the near future.
-        #
-        # Likewise for the other commands we barr here.
-        found_barred = False
-        for name, test in LLDB_EXCLUSIVE:
-            if not test(bits[0]):
-                continue
+            if not action._prompt_silent:
+                print(f"{PROMPT}{action._command}")
 
-            print(
-                message.error(
-                    f"The '{name}' command is not supported. Use the 'lldb' command to enter LLDB mode and try again."
-                )
-            )
+            output = BytesIO()
```

Comment:
can we just do?
```suggestion
            output = sys.stdout.buffer if action._capture else BytesIO()
```

Reply:
Or we can implement `action._capture_file`? and default it should be `sys.stdout.buffer`?

Reply:
MyPy yells at us if we do, since apparently `sys.stdout.buffer` doesn't inherit from `BufferedIOBase` 🥲

Of course, we can also just ignore typing, entirely, and go with `Any` in this case. I'd be perfectly fine with that. I just chose to do it this way cuz I figured in most cases it wouldn't really make much of a difference either way.

Reply:
> MyPy yells at us if we do

can we just do `Union[BufferedIOBase, stdout_buffer_type]` ?

Reply:
🤔 And I'm think about this new `typing.Protocol` just for `write` method

Reply:
If I remember correctly, `stdout_buffer_type` is `Any`

Reply:
@mbrla0 `typing.BinaryIO` works fine with both `sys.stdout.buffer` and io.BytesIO

Reply:
Well, fair enough, then

Reply:
Should be better now

Reply:
👍